packages/realm-react/src/RealmProvider.tsx

```diff
@@ -161,6 +193,47 @@ export function createRealmProvider(
   };
 }
 
+/**
+ * Generates a `RealmProvider` which is either based on a configuration
+ * or based on a realm, depending on its props.
+ * @param RealmContext - The context that will contain the Realm instance
+ * @returns a RealmProvider component that provides context to all context hooks
+ */
+export function createDynamicRealmProvider(RealmContext: React.Context<Realm | null>): DynamicRealmProvider {
+  return ({ realm, children, ...configurationProps }) => {
+    if (realm) {
+      if (!isEqual(configurationProps, {})) {
```

Comment:
@kraenhansen @elle-j @kneth
I think this is the last thing worth taking a look at. Added this so there can be runtime validation and a human-friendly error in case they pass both types of props. If no comments about that, should be ready to merge.

Reply:
This could also be simplified (limiting dependence on the lodash dependency), although I don't have a strong opinion here:

```suggestion
      if (Object.keys(configurationProps).length === 0) {
```